### PR TITLE
GEODE-5626: Reject getInitialValues=true in registerAllKeys when regi…

### PR DESCRIPTION
**Pull request by pdxcodemonkey**
…on is not caching.

    - Refactor Cluster class so we can apply locators separately.
    - Clean up server directories at test startup.
    - Add test cases for caching proxy and proxy regions with getInitialValues set to true

Co-authored-by: Ivan Godwin <igodwin@pivotal.io>

### CODE REVIEW

File: cppcache/integration-test-2/RegisterKeysTest.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    region2->registerKeys(keys, false, true);
+
+    auto&& entryAfterGet = std::dynamic_pointer_cast<CacheableInt16>(
+        region2->getEntry("one")->getValue());
+    ASSERT_NE(entryAfterGet, nullptr);
+    ASSERT_EQ(entryAfterGet->value(), 1);
+  }
+}
+
+/**
```
Comment (pivotal-jbarrett) [inline cppcache/integration-test-2/RegisterKeysTest.cpp]: Either add documentation to the test or don't but don't add empty documentation.

File: cppcache/integration-test-2/framework/Cluster.h [hunk 2]
```diff
@@ -182,6 +182,8 @@ class Cluster {
         initialServers_(initialServers.get()) {
     jmxManagerPort_ = Framework::getAvailablePort();
 
+    // Clean up from previous run, if any, to ensure a clean server
```
Comment (pivotal-jbarrett) [inline cppcache/integration-test-2/framework/Cluster.h]: This comment is redundant to the statement below. Code should be expressive such that it doesn't need comments. Code needing preceding comments should be refactored to be more expressive.

File: cppcache/integration-test-2/framework/Cluster.h [hunk 3]
```diff
@@ -208,32 +210,38 @@ class Cluster {
 
   void stop();
 
+  void removeServerDirectory() {
+    boost::filesystem::path serverDir = boost::filesystem::current_path().concat("/" + name_);
```
Comment (pivotal-jbarrett) [inline cppcache/integration-test-2/framework/Cluster.h]: Why do you need to `concat` and `+` to get a sub path?

File: cppcache/integration-test-2/framework/Cluster.h [hunk 4]
```diff
@@ -208,32 +210,38 @@ class Cluster {
...
+    boost::filesystem::remove_all(serverDir);
```
Comment (pivotal-jbarrett) [inline cppcache/integration-test-2/framework/Cluster.h]: Will this not operate relative to the current working directory?

File: cppcache/integration-test-2/framework/Cluster.h (hunk 2 above)
Comment (pivotal-jbarrett) [reply cppcache/integration-test-2/framework/Cluster.h]: I would like to see an integration test written to test this change.
I would also like to see one that asserts that `GfshCommand::execute()` throws an exception when `gfsh` exits non-zero.
